docs/source/devel/design.rst

```diff
@@ -1,19 +1,214 @@
-``rstuf-repo-worker``) and one tool (``rstuf-cli``)
+###################
+Architecture Design
+###################
 
-.. image:: /_static/2_2_rstuf.png
+The principles
+##############
 
-By design, it helps scalability and reliability for complex and
-distributed deployment. As the example below
 
+   * RSTUF uses `The Update Framework (TUF) <http://www.theupdateframework.io>`_.
 
-.. image:: /_static/2_3_rstuf.png
+     - It enables RSTUF to be artifact agnostic.
 
+   * RSTUF is easy to deploy.
 
-Components/Tools
-================
+   * RSTUF is API first designed.
 
-- `repository-service-tuf-api <https://repository-service-tuf.readthedocs.io/projects/rstuf-api/>`_
-- `repository-service-tuf-worker <https://repository-service-tuf.readthedocs.io/projects/rstuf-worker/>`_
-- `repository-service-tuf-cli <https://repository-service-tuf.readthedocs.io/projects/rstuf-cli/>`_
+     - RSTUF is language agnostic, making any language use/integrate easily.
+
+   * RSTUF is process agnostic.
+
+     - Adding/removing artifacts goes with the existing releasing/publishing
+       process.
+
+   * RSTUF focuses on scalability and Metadata consistency.
+
+   * RSTUF guides users on TUF processes.
+
+
+The architecture design principles:
+###################################
+
+The Repository Service for TUF (RSTUF) has two services
+(``repository-service-tuf-api``, ``repository-service-tuf-worker``) and one
+command line tool (``repository-service-cli``), as described in
+:ref:`guide/general/introduction:RSTUF Components`.
+
+.. note::
+    Other services can be added as optional or required
+
+RSTUF also requires some third-party services described in
+:ref:`guide/deployment/planning/deployment:Required Infrastructure Services`.
+
+.. image:: /_static/2_1_rstuf.png
+
+The below definitions allow RSTUF API and RSTUF Worker scalability.
+
+RSTUF is Asynchronous
+=====================
+
+    * Every API request is a task.
+    * RSTUF centralizes all tasks in the Message :ref:`devel/design:Message Queue`.
+    * RSTUF stores all task results in the :ref:`devel/design:Backend Result`.
+    * :ref:`devel/design:Repository Service TUF API` **publishes** RSTUF tasks.
+    * :ref:`devel/design:Repository Service TUF Worker` **consumes** RSTUF tasks.
+
+RSTUF Repository Settings/Configuration
+=======================================
+
+    * :ref:`guide/general/introduction:RSTUF Components` are RSTUF
+      **stateless** about the Repository Settings, relying on
+      :ref:`devel/design:RSTUF settings`.
+
+    * :ref:`devel/design:RSTUF settings` stores in
+      :ref:`devel/design:Redis` using `Dynaconf <https://www.dynaconf.com>`_.
+
+    * :ref:`devel/design:Repository Service TUF Worker` **writes** settings
+    * :ref:`devel/design:Repository Service TUF API` **reads** settings.
+
+        .. note::
+          A unique exception is when a bootstrap process fails and the API
+          **writes** it back to ``None``.
```

Comment:
This note looks a little bit out of place:
<img width="387" alt="image" src="https://github.com/repository-service-tuf/repository-service-tuf/assets/16246778/db80e476-3cf6-4dfc-9aad-a81062bc0c6b">

Is there a better place for this?

Reply:
The place is correct. I will reformulate the sentence.

Reply:
I rephrased the sentence and gave the full context of the exception.